src/jdk.jartool/share/classes/sun/security/tools/jarsigner/Main.java

```diff
@@ -1965,9 +1965,9 @@ void signJar(String jarName, String alias)
         builder.setProperty("sectionsOnly", Boolean.toString(!signManifest));
         builder.setProperty("internalSF", Boolean.toString(!externalSF));
 
-        FileOutputStream fos = null;
+        OutputStream os = null;
         try {
-            fos = new FileOutputStream(signedJarFile);
```

Comment:
Hello Max, what you suggest sounds fine to me. I've updated the PR to follow this suggestion. The only difference now would be that in the previous proposed patch if any exception happened during the jarsiging the `BufferedOutputStream` would be (flushed and) closed from the finally block of the main method. However, in the newer version, there can be a case that if an exception occurs then the underlying BufferedOutputStream, constructed in the `JarSigner` class may not be flushed/closed, but that I think should be fine, because we anyway delete the temporary signed jar file, in the main method, if any exception occurs.

With this new change, I again ran the test against a 3GB and 6GB file and the numbers continue to show the improvement. I'll trigger the tier testing too.